qiita_pet/handlers/preprocessing_handlers.py

```diff
@@ -13,19 +13,12 @@ class PreprocessHandler(BaseHandler):
     def post(self):
         study_id = int(self.get_argument('study_id'))
         prep_template_id = int(self.get_argument('prep_template_id'))
-        raw_data = PrepTemplate(prep_template_id).artifacts
+        raw_data = PrepTemplate(prep_template_id).artifact
         param_id = int(self.get_argument('preprocessing_parameters_id'))
 
-        # Get the preprocessing parameters
-        if raw_data.filetype in ('FASTQ', 'per_sample_FASTQ'):
-            param_constructor = PreprocessedIlluminaParams
-        elif raw_data.filetype in ('FASTA', 'SFF'):
-            param_constructor = Preprocessed454Params
-        else:
-            raise ValueError('Unknown filetype')
+        user_id = self.current_user.id
 
-        job_id = submit(self.current_user.id, preprocessor, study_id,
-                        prep_template_id, param_id, param_constructor)
+        job_id = submit(user_id, preprocessor, user_id, raw_data.id, param_id)
```

Comment:
That is part of the execution environment that we have in place, and it is using moi. The first user_id is used in moi to identify which user is launching the job, so we can use the websockets to communicate back to the interface. The second user_id is actually a parameter of the function `preprocessor`.

In short, the parameters of the `submit` function is a user identifier to know who is launching the job, the function to be executed and the `args` and `kwargs` of such function.